src/meds_torch/models/eic_forecasting.py

```diff
@@ -162,31 +223,163 @@ def compute(self):
+
+
+    # >>>
+    # >>> # Test workflow 1: Autoregressive training
+    # >>> model = EicForecastingModule(cfg)
+    # >>> loss = model.training_step(batch)
+    # >>> assert loss.isfinite().all()
+    # >>>
+    # >>> # Test workflow 2: Data generation
+    # >>> cfg.num_samples = 2  # Enable generation
+    # >>> model = EicForecastingModule(cfg)
+    # >>>
+    # >>> # Test generation
+    # >>> output = model.forward(batch)
+    # >>> print(f"Generated sequences shape: {output['GENERATE//0']['code'].shape}")
+    # Generated sequences shape: torch.Size([2, 3])
+    # >>> # Test workflow 3: Zero-shot prediction
+    # >>> cfg.zero_shot_labeler = lambda x: torch.tensor([0.7, 0.3])
+    # >>> model = EicForecastingModule(cfg)
+    # >>> output = model.forward(batch)
+    # >>> print(f"Zero-shot predictions shape: {output[MODEL_PRED_PROBA_KEY].shape}")
+    # Zero-shot predictions shape: torch.Size([2])
+    # >>>
+
+
+    >>> # Test generation with real model
+    >>> cfg.num_samples = 2  # Enable generation
+    >>> B, S, L = 2, 5, 8 # [batch_size, input_sequence_length, token_dim]
+    >>> vocab_size = 4
+    >>> max_seq_len = 10
+    >>> cfg.temperature = 1000.0 # Raise the temperature to randomize the predictions
+    >>> model = EicForecastingModule(cfg)
+    >>> _ = torch.manual_seed(42)
+    >>> model.model = instantiate({
+    ...     '_target_': "meds_torch.models.components.transformer_decoder.TransformerDecoderModel.initialize",
+    ...     'token_dim': L,
+    ...     'vocab_size': vocab_size,
+    ...     'max_seq_len': max_seq_len,
+    ...     'get_last_token': True,
+    ...     'token_emb': None,
+    ...     'generation_budget': {
+    ...         '_target_': ("meds_torch.models.components.transformer_decoder."
+    ...                      "GenerationBudget.from_time_len"),
+    ...         "value": .5,
+    ...     },
+    ...     'model': {
+    ...         '_target_': 'x_transformers.TransformerWrapper',
+    ...         'num_tokens': vocab_size,
+    ...         'max_seq_len': max_seq_len,
+    ...         'attn_layers': {
+    ...             '_target_': 'x_transformers.Decoder',
+    ...             'dim': L,
+    ...             'depth': 2,
+    ...             'heads': 2,
+    ...         }
+    ...     }
+    ... })
+    >>>
+    >>> # Create input batch
+    >>> output = model.forward(batch)
+    >>> codes = output['GENERATE//0']['code']
+    >>> codes.shape[0] == 2
+    True
+    >>> codes.shape[1] >= 1
+    True
+    >>> # Note that the time token is code/vocab_index 2 in the medadata
+    >>>  # check that 1 time token was generated for the shortest time trajectory
+    >>> (codes == 2).sum(dim=1).min().item()
+    1
+    >>> temp_file.close()
+    """
 
     def __init__(self, cfg: DictConfig):
         super().__init__(cfg)
-        if not isinstance(self.model, AUTOREGRESSIVE_MODELS):
+        if not isinstance(self.model, AUTOREGRESSIVE_MODELS + (DummyModel,)):
```

Comment:
You don't want to have to modify your code for these sorts of testing settings... Is there any way to put `DummyModel` in the list of `AUTOREGRESSIVE_MODELS` somehow, either in general if appropriate or only during doctests? Similarly for the encoder